subsys/bluetooth/mesh/proxy_srv.c

```diff
@@ -884,10 +886,23 @@ static struct bt_gatt_attr proxy_attrs[] = {
 };
 
 static struct bt_gatt_service proxy_svc = BT_GATT_SERVICE(proxy_attrs);
+static void svc_reg_work_handler(struct k_work *work);
+static struct k_work_delayable svc_reg_work = Z_WORK_DELAYABLE_INITIALIZER(svc_reg_work_handler);
 
 static void svc_reg_work_handler(struct k_work *work)
 {
-	(void)bt_gatt_service_register(&proxy_svc);
+	int err;
+
+	err = bt_gatt_service_register(&proxy_svc);
+	if (err == -EINVAL) {
+		/* settings_load() didn't finish yet. Try again. */
```

Comment:
What if gatt service cannot be registered for another reason? Will they forever invoke each other?